### ISSUE TITLE
Push restrictions aren't applied properly on branches

### ISSUE THREAD

**Issue by GergelyKalmar** [integrations/terraform-provider-github#670]
### Terraform Version
```
Terraform v0.14.5
+ provider registry.terraform.io/integrations/github v4.3.1
```

### Affected Resource(s)
- github_branch_protection

### Terraform Configuration Files
```hcl
resource "github_branch_protection" "main" {
  repository_id = github_repository.repo.name
  pattern = "main"
  push_restrictions = []
}
```

### Expected Behavior
The "Restrict who can push to matching branches" option should be turned on for the given branch (with no team or user added).

### Actual Behavior
The option is unchanged.

### Steps to Reproduce
1. `terraform plan`
2. `terraform apply`

**Comment by jcudit** [comment]
Thanks for reporting this.  Not a fix, but curious if the `github_branch_protection_v3` resource works for your use case?  Will track this otherwise during our next batch of `github_branch_protection` fixes.

**Comment by suever** [comment]
> Thanks for reporting this. Not a fix, but curious if the `github_branch_protection_v3` resource works for your use case? Will track this otherwise for our next batch of `github_branch_protection` fixes.

Not the OP, but setting `restrictions` to be empty in the `github_branch_protection_v3` has the expected behavior 

```
resource "github_branch_protection_v3" "main" {
  repository = github_repository.repo.name
  branch = "main"
  restrictions = {}
}
```

**Comment by GergelyKalmar** [comment]
Indeed, it seems to be working as expected with `v3`. That should do as a workaround for now, thanks for the tip!

**Comment by spangaer** [comment]
@james-callahan 

Create a dummy empty secret team and use it's node ID in the list, that seems to have the wanted effect

**Comment by GergelyKalmar** [comment]
I think the issue still persists, I'd rather have it remain active. Having multiple versions of `github_branch_protection` is not particularly elegant.

**Comment by marcaurele** [comment]
The issue still persists, we are affected by this and have to use a workaround by duplicating the accesses of the default users/teams.

**Comment by kfcampbell** [comment]
Agreed that having multiple branch protection resources is not an elegant solution. PRs are appreciated if folks have the time and inclination to fix this behavior.
